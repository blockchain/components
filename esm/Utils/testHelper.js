import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}