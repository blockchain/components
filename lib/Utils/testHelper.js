"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}