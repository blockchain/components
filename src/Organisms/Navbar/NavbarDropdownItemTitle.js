import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Text from '../../Atoms/Typography/Text'

const Wrapper = styled(Text).attrs({
  size: '1rem',
  weight: 600,
})`
  color: ${(props) => props.theme['navbarDropdownTitleMobileColor']};

  @media (min-width: 48rem) {
    color: ${(props) => props.theme['navbarDropdownTitleColor']};
  }
`

const NavbarDropdownItemTitle = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

NavbarDropdownItemTitle.propTypes = {
  children: PropTypes.node,
}

export default NavbarDropdownItemTitle
