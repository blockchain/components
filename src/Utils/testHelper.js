import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
