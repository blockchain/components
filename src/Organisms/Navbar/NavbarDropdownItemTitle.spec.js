import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import NavbarDropdownItemTitle from './NavbarDropdownItemTitle'

describe('NavbarDropdownItemTitle', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<NavbarDropdownItemTitle />)

    expect(tree).toMatchSnapshot()
  })
})
