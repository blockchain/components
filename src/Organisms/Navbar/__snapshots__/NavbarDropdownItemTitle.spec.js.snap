// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarDropdownItemTitle should render correctly 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  color: #FFF;
}

@media (min-width:48rem) {
  .c0 {
    color: #3558A8;
  }
}

<NavbarDropdownItemTitle>
  <Styled(Text)>
    <Text
      capitalize={false}
      className="c0"
      italic={false}
      opacity={1}
      size="1rem"
      uppercase={false}
      weight={600}
    >
      <styled.span
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="1rem"
        uppercase={false}
        weight={600}
      >
        <span
          className="c0 c1"
          opacity={1}
          size="1rem"
        />
      </styled.span>
    </Text>
  </Styled(Text)>
</NavbarDropdownItemTitle>
`;
